Targets - swap 2 target objects to be files & 1 target file to be an object

lindsayplatt opened the pull request:
Working on #7 - I changed the combined site data object target `site_data` and the plot-ready object target `site_data_plot` into `.rds` file targets. Additionally, I changed the `1_fetch/out/site_info.csv` file target to be an object instead.

I do not have anything in `build/status` to comment here.

lindsayplatt committed:
turn site_data and site_data_plot into target files

lindsayplatt committed:
turn site info target file into a target object

aappling-usgs commented:
> I do not have anything in build/status to comment here.

Huh, I hadn't noticed before, but the language in the issue is unclear. It says something like

> paste your build status

which I think is the instruction you're interpreting as above. But I think what @jread-usgs meant was to paste console output such as this (probably after calling `scmake()` with no arguments):
```
> scmake('site_data')
Starting build at 2020-06-24 11:09:41
<  MAKE > site_data
[    OK ] site_data
Finished build at 2020-06-24 11:09:41
Build completed in 0.00 minutes
```

jordansread commented:
yes, that is what I meant...whoops

aappling-usgs commented on 1_fetch/src/get_nwis_data.R:
```diff
@@ -17,10 +17,11 @@ download_nwis_data <- function(site_nums = c("01427207", "01432160", "01435000",
 }
 
-nwis_site_info <- function(fileout, site_data){
+nwis_site_info <- function(fileout, site_data_file){
```
You don't need this `fileout` argument anymore, right?

aappling-usgs reviewed (approved):
Looks great, no surprise, though I think you missed removing one argument

aappling-usgs commented:
Go ahead and merge whenever you're ready, @lindsayplatt

lindsayplatt commented:
Got it!

### `scmake()` after commit where I changed objects into file targets

```
> scmake()
Starting build at 2020-06-24 10:59:41
<  MAKE > all
[    OK ] 1_fetch/out/site_data_01427207.csv
[    OK ] 1_fetch/out/site_data_01432160.csv
[    OK ] 1_fetch/out/site_data_01435000.csv
[    OK ] 1_fetch/out/site_data_01436690.csv
[    OK ] 1_fetch/out/site_data_01466500.csv
[ BUILD ] 2_process/out/site_data.rds        |  combine_and_process_data("2_process/...
[  READ ]                                    |  # loading packages
Parsed with column specification:
cols(
  agency_cd = col_character(),
  site_no = col_character(),
  dateTime = col_datetime(format = ""),
  X_00010_00000 = col_double(),
  X_00010_00000_cd = col_character(),
  tz_cd = col_character()
)
Parsed with column specification:
cols(
  agency_cd = col_character(),
  site_no = col_character(),
  dateTime = col_datetime(format = ""),
  X_00010_00000 = col_double(),
  X_00010_00000_cd = col_character(),
  tz_cd = col_character()
)
Parsed with column specification:
cols(
  agency_cd = col_character(),
  site_no = col_character(),
  dateTime = col_datetime(format = ""),
  X_00010_00000 = col_double(),
  X_00010_00000_cd = col_character(),
  tz_cd = col_character()
)
Parsed with column specification:
cols(
  agency_cd = col_character(),
  site_no = col_character(),
  dateTime = col_datetime(format = ""),
  X_00010_00000 = col_double(),
  X_00010_00000_cd = col_character(),
  tz_cd = col_character()
)
Parsed with column specification:
cols(
  agency_cd = col_character(),
  site_no = col_character(),
  dateTime = col_datetime(format = ""),
  X_00010_00000 = col_double(),
  X_00010_00000_cd = col_character(),
  tz_cd = col_character()
)
[ BUILD ] 1_fetch/out/site_info.csv          |  nwis_site_info(fileout = "1_fetch/ou...
[ BUILD ] 2_process/out/site_data_plot.rds   |  prepare_data_for_plot("2_process/out...
Parsed with column specification:
cols(
  .default = col_character(),
  lat_va = col_double(),
  long_va = col_double(),
  dec_lat_va = col_double(),
  dec_long_va = col_double(),
  district_cd = col_double(),
  state_cd = col_double(),
  map_scale_fc = col_double(),
  alt_va = col_double(),
  alt_acy_va = col_double(),
  basin_cd = col_logical(),
  topo_cd = col_logical(),
  construction_dt = col_logical(),
  inventory_dt = col_double(),
  drain_area_va = col_double(),
  contrib_drain_area_va = col_logical(),
  reliability_cd = col_logical(),
  nat_aqfr_cd = col_logical(),
  aqfr_cd = col_logical(),
  aqfr_type_cd = col_logical(),
  well_depth_va = col_logical()
  # ... with 3 more columns
)
See spec(...) for full column specifications.
[ BUILD ] 3_visualize/out/figure_1.png       |  plot_nwis_timeseries(fileout = "3_vi...
[ ----- ] all
Finished build at 2020-06-24 10:59:52
Build completed in 0.20 minutes
```

### `scmake()` after I changed the file target into an object target

```
> scmake()
Starting build at 2020-06-24 11:02:06
<  MAKE > all
[    OK ] 1_fetch/out/site_data_01427207.csv
[    OK ] 1_fetch/out/site_data_01432160.csv
[    OK ] 1_fetch/out/site_data_01435000.csv
[    OK ] 1_fetch/out/site_data_01436690.csv
[    OK ] 1_fetch/out/site_data_01466500.csv
[    OK ] 2_process/out/site_data.rds
[ BUILD ] site_info                          |  site_info <- nwis_site_info(site_dat...
[  READ ]                                    |  # loading packages
[ BUILD ] 2_process/out/site_data_plot.rds   |  prepare_data_for_plot("2_process/out...
[    OK ] 3_visualize/out/figure_1.png
[ ----- ] all
Finished build at 2020-06-24 11:02:08
Build completed in 0.04 minutes
```

lindsayplatt committed:
no longer need fileout arg

lindsayplatt commented:
Build status after removing unused argument to `nwis_site_info`

```
> scmake()
Starting build at 2020-06-24 11:18:33
<  MAKE > all
[    OK ] 1_fetch/out/site_data_01427207.csv
[    OK ] 1_fetch/out/site_data_01432160.csv
[    OK ] 1_fetch/out/site_data_01435000.csv
[    OK ] 1_fetch/out/site_data_01436690.csv
[    OK ] 1_fetch/out/site_data_01466500.csv
[    OK ] 2_process/out/site_data.rds
[ BUILD ] site_info                          |  site_info <- nwis_site_info(site_dat...
[ BUILD ] 2_process/out/site_data_plot.rds   |  prepare_data_for_plot("2_process/out...
[    OK ] 3_visualize/out/figure_1.png
[ ----- ] all
Finished build at 2020-06-24 11:18:35
Build completed in 0.04 minutes
```
